modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTController.scala

```diff
@@ -22,10 +22,85 @@ import squants.time.Time
 final case class GHOSTController[F[_]: Sync](ghostClient: GHOSTClient[F],
                                              gdsClient: GDSClient) {
   import GHOSTController._
-  val log: Logger = getLogger
+  val Log: Logger = getLogger
+
+  private sealed abstract class BundleConfig(val configName: String)
+  private final case object StandardBundleConfig extends BundleConfig("lo")
+  private final case object HighResBundleConfig extends BundleConfig("hi")
+
+  private sealed abstract class IFUNum(val ifuNum: Int) {
+    val ifuStr = s"ghost:cc:cu:ifu$ifuNum"
+  }
+  private final case object IFU1 extends IFUNum(1)
+  private final case object IFU2 extends IFUNum(ifuNum = 2)
+
+  private sealed abstract class IFUTargetType(val targetType: Int)
+  private object IFUTargetType {
+    def determineType(name: Option[String]): IFUTargetType = name match {
+      case None        => IFUTargetNone
+      case Some("Sky") => IFUTargetSky
+      case _           => IFUTargetObject
+    }
+  }
+  private final case object IFUTargetNone extends IFUTargetType(0)
+  private final case object IFUTargetSky  extends IFUTargetType(1)
+  private final case object IFUTargetObject extends IFUTargetType(2)
+
```

Comment:
Is there any benefit in these being private? We usually don't bother.

I don't think these need to be nested classes. This creates path-dependent types which aren't what we want. So I would move them to the companion object. 

Also by convention we place the constructors for an ADT inside the companion. 

So:

```scala
object GHOSTController {
  ...
  sealed abstract class IFUTargetType(val targetType: Int)
  object IFUTargetType {
    case object None extends IFUTargetType(0)
    case object Sky  extends IFUTargetType(1)
    ...
   }

}
```

Then your class looks like

```scala
final case class GHOSTController... {
  import GHOSTController._

  // in here you can now say
  IFUTargetType.None

}
```

Reply:
There's no particular reason to make them private. I guess it's a product of the C++ liquifying my brain. 🤯

Okay, those changes are easy enough to make and make sense; having the case objects in the companion object to the sealed class / trait also allows for much simpler names to be used.

Thanks for the tips... working on old Scala code and Java code in the old OCS has me lagging a bit.